client/my-sites/theme/main.jsx

```diff
@@ -797,6 +808,7 @@ const ThemeSheetWithOptions = ( props ) => {
 	return (
 		<ConnectedThemeSheet
 			{ ...props }
+			demo_uri={ 'Storefront' === themeName ? storefrontDemoUri : demo_uri }
```

Comment:
What do you think about defining the demo_uri property using [the getThemeDemoUrl selector](https://github.com/Automattic/wp-calypso/blob/79c4583a523a3d6deed21f5ec79cfbdd03fab28a/client/state/themes/selectors/get-theme-demo-url.js#L16)? This way we would only define the storefrontDemoUri constant one time.

I pushed the suggestion up to this branch (untested): update/storefront-demo-link_danielsmithmichigan-suggestion

Reply:
Good call! I honestly wondered if I should have added the logic in `getCanonicalTheme`, but then it feels wrong to mess with, uh, canon. 🤷‍♀️ I've swapped the passed prop for the selector in this case.